test/WebJobs.Script.Tests.Integration/Host/StandbyManagerTests.cs

```diff
@@ -110,109 +101,112 @@ public async Task StandbyMode_EndToEnd_LinuxContainer()
-                Assert.Equal(1, logLines.Count(p => p.Contains("Creating StandbyMode placeholder function directory")));
-                Assert.Equal(1, logLines.Count(p => p.Contains("StandbyMode placeholder function directory created")));
-                Assert.Equal(2, logLines.Count(p => p.Contains("Starting Host (HostId=placeholder-host")));
-                Assert.Equal(2, logLines.Count(p => p.Contains("Host is in standby mode")));
-                Assert.Equal(2, logLines.Count(p => p.Contains("Executed 'Functions.WarmUp' (Succeeded")));
-                Assert.Equal(1, logLines.Count(p => p.Contains("Validating host assignment context")));
-                Assert.Equal(1, logLines.Count(p => p.Contains("Starting Assignment")));
-                Assert.Equal(1, logLines.Count(p => p.Contains("Applying 1 app setting(s)")));
-                Assert.Equal(1, logLines.Count(p => p.Contains($"Extracting files to '{_webHostOptions.ScriptPath}'")));
-                Assert.Equal(1, logLines.Count(p => p.Contains("Zip extraction complete")));
-                Assert.Equal(1, logLines.Count(p => p.Contains("Triggering specialization")));
-                Assert.Equal(1, logLines.Count(p => p.Contains("Starting host specialization")));
-                Assert.Equal(1, logLines.Count(p => p.Contains($"Starting Host (HostId={_expectedHostId}")));
-                Assert.Contains("Node.js HttpTrigger function invoked.", logLines);
-
-                // verify cold start log entry
-                var coldStartLog = _loggerProvider.GetAllLogMessages().FirstOrDefault(p => p.Category == ScriptConstants.LogCategoryHostMetrics);
-                JObject coldStartData = JObject.Parse(coldStartLog.FormattedMessage);
-                Assert.Equal("Dynamic", coldStartData["sku"]);
-                Assert.True((int)coldStartData["dispatchDuration"] > 0);
-                Assert.True((int)coldStartData["functionDuration"] > 0);
-            }
+
+            var environment = new TestEnvironment(vars);
+
+            InitializeTestHost("Linux", environment);
+
+            await VerifyWarmupSucceeds();
+            await VerifyWarmupSucceeds(restart: true);
+
+            // now specialize the site
+            await Assign(encryptionKey);
+
+            await TestHelpers.Await(() => environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsitePlaceholderMode) == "0");
+
+            // immediately call a function - expect the call to block until
+            // the host is fully specialized
+            // the Unauthorized is expected since we havne't specified the key
+            // it's enough here to ensure we don't get a 404
+            var request = new HttpRequestMessage(HttpMethod.Get, $"api/httptrigger");
+            request.Headers.Add(ScriptConstants.AntaresColdStartHeaderName, "1");
+            var response = await _httpClient.SendAsync(request);
+            Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
```

Comment:
This test currently fails right here, which I'm investigating. But wanted to get the other test running ASAP since it would have caught a couple of issues.

Reply:
This test now passes.